### PR TITLE
Hia 505 person risk pull back one field

### PR DESCRIPTION
**Pull request by AprilMD**
## Context

Several risk assessment endpoints are coming up. We want to make implementing these endpoints easier and as a result we begin by taking a single field from an upstream api. All of these endpoints will use ARN and as a result this PR includes the ARN gateway.

Worth noting that this PR does not include smoke tests as the api docs for ARN are not public. We are requesting/exploring making them public in this [ticket](https://dsdmoj.atlassian.net/jira/software/c/projects/HIA/boards/1056?modal=detail&selectedIssue=HIA-504)

## Changes proposed in this PR

- Creation of a ARN gateway
- Implement RiskPredictorScore Models, Service and controller

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/gateways/assessrisksandneeds/GetRiskPredictorScoresForPersonTest.kt [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        assessRisksAndNeedsApiMockServer.stubGetRiskPredictorScoresForPerson(
+          crn,
+          """
+            [
+                {
+                    "generalPredictorScore": {
+                        "ogpRisk": 0
+                    }
+                }
+            ]
```
Comment (emileswarts) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/gateways/assessrisksandneeds/GetRiskPredictorScoresForPersonTest.kt:49]: line spacing of 2 spaces should be enough here

**Review by emileswarts — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/models/UpstreamApi.kt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 package uk.gov.justice.digital.hmpps.hmppsintegrationapi.models
 
 enum class UpstreamApi {
-  NOMIS, PRISONER_OFFENDER_SEARCH, PROBATION_OFFENDER_SEARCH, NDELIUS
+  NOMIS, PRISONER_OFFENDER_SEARCH, PROBATION_OFFENDER_SEARCH, NDELIUS, ARN
```
Comment (chubberlisk) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/models/UpstreamApi.kt:4]:
```suggestion
  NOMIS, PRISONER_OFFENDER_SEARCH, PROBATION_OFFENDER_SEARCH, NDELIUS, ASSESS_RISKS_AND_NEEDS
```

Could we use the full name of the API like we've done with the other APIs to remove the cognitive load (and me thinking Amazon Resource Name)? 😝

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/services/GetRiskPredictorScoresForPersonServiceTest.kt [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+      )
+
+      val response = getRiskPredictorScoresForPersonService.execute(pncId)
+
+      response.data.shouldBe(riskPredictors)
+    }
+
+    describe("when an upstream API returns an error") {
+
+      xdescribe("when a person cannot be found by pnc ID in probation offender search") {
```
Comment (chubberlisk) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/services/GetRiskPredictorScoresForPersonServiceTest.kt:86]: Are these tests skipped for a reason?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/services/GetRiskPredictorScoresForPersonServiceTest.kt [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+)
+internal class GetRiskPredictorScoresForPersonServiceTest(
+  @MockBean val assessRisksAndNeedsGateway: AssessRisksAndNeedsGateway,
+  @MockBean val getPersonService: GetPersonService,
+  private val getRiskPredictorScoresForPersonService: GetRiskPredictorScoresForPersonService,
+) : DescribeSpec(
+  {
+    val pncId = "1234/56789B"
+    val nomisNumber = "Z99999ZZ"
+    val crn = "X123456"
```
Comment (chubberlisk) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/services/GetRiskPredictorScoresForPersonServiceTest.kt:34]: I think we should refer to CRN to `deliusCrn` everywhere as that's how we're exposing it and to also be consistent with `nomisNumber`.

See https://ministryofjustice.github.io/hmpps-integration-api-docs/documentation/api/index.html#schema-identifiers.

image

**Review by chubberlisk — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/models/UpstreamApi.kt (hunk 2 above)
Comment (popey2700) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/models/UpstreamApi.kt:4]: Yeah good thinking, will change!

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/services/GetRiskPredictorScoresForPersonServiceTest.kt (hunk 3 above)
Comment (popey2700) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/services/GetRiskPredictorScoresForPersonServiceTest.kt:86]: Yup, the functionality for when someone can't be found isn't in the personService correctly yet. So these tests will be relevant for once that functionality has been added to the personService

Comment (chubberlisk) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/services/GetRiskPredictorScoresForPersonServiceTest.kt:86]: Cool beans, thanks for the clarification!

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/services/GetRiskPredictorScoresForPersonServiceTest.kt (hunk 4 above)
Comment (popey2700) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/services/GetRiskPredictorScoresForPersonServiceTest.kt:34]: Cool, good call

**Pull request merged**